icu4c/source/common/restrace.cpp

```diff
@@ -28,13 +28,33 @@ void ResourceTracer::trace(const char* resType) const {
     icu::CharString resPath;
     getResPath(resPath, status);
 
-    UTRACE_DATA3(UTRACE_VERBOSE, "(%s) %s @ %s",
+    // The longest type ("intvector") is 9 chars
+    icu::CharString format;
+    for (int32_t i=uprv_strlen(resType); i<9; i++) {
+        format.append(' ', status);
```

Comment:
I would use a nine-space string literal "         " and append a prefix of that with the desired padding length, rather than the loop.